gnat2goto/driver/arrays.adb

```diff
@@ -228,24 +227,28 @@ package body Arrays is
          while Present (Pos_Iter) loop
             declare
                Expr : constant Irep := Do_Expression (Pos_Iter);
+               Pos_Constant : constant Irep :=
```

Comment:
:pick: The declaration of variables like `Pos_Iter` is way too far removed from their usage for my taste; May be worth moving the subprocedures out into separate procedures entirely, or at least declare the variables that are needed in the main body directly above it rather than above the procedures